ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx

```diff
@@ -17,65 +22,180 @@ import {
       (account: InternalAccount) => allowedAccountTypes.includes(account.type),
     );
-  }, [accounts, internalAccounts, allowedAccountTypes]); // Add allowedAccountTypes to dependency array
+  }, [accounts, internalAccounts, allowedAccountTypes]);
+
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+  const subjects = useSelector(getPermissionSubjects);
+  const connectedAccounts = useSelector((state: any) =>
+    getOrderedConnectedAccountsForConnectedDapp(state, activeTabOrigin).filter(
+      (account: InternalAccount) => isEvmAccountType(account.type),
+    ),
+  );
+
+  const connectedAccountsAddresses = connectedAccounts.map(
+    (account: InternalAccount) => account.address,
+  );
+
+  const [selectedAccounts, setSelectedAccounts] = useState<string[]>(
+    connectedAccountsAddresses,
+  );
+
+  const handleAccountClick = (address: string) => {
+    const index = selectedAccounts.indexOf(address);
+    let newSelectedAccounts: string[];
+
+    if (index === -1) {
+      newSelectedAccounts = [...selectedAccounts, address];
+    } else {
+      newSelectedAccounts = selectedAccounts.filter(
+        (account) => account !== address,
+      );
+    }
+    setSelectedAccounts(newSelectedAccounts);
+  };
+
+  const disconnectAllAccounts = () => {
+    const subject = (subjects as SubjectsType)[activeTabOrigin];
+    if (subject) {
+      const permissionMethodNames = Object.values(subject.permissions).map(
+        ({ parentCapability }: { parentCapability: string }) =>
+          parentCapability,
+      ) as string[];
+      if (permissionMethodNames.length > 0) {
+        const permissionsRecord: Record<string, string[]> = {
+          [activeTabOrigin]: permissionMethodNames,
+        };
+
+        dispatch(
+          removePermissionsFor(
+            permissionsRecord as Record<string, NonEmptyArray<string>>,
+          ),
+        );
+      }
+    }
+  };
+
+  const managePermittedAccounts = (
+    selectedAccounts: string[],
+    connectedAccountsAddresses: string[],
+    activeTabOrigin: string,
+  ) => {
+    const removedElements = connectedAccountsAddresses.filter(
+      (account) => !selectedAccounts.includes(account),
+    );
+
+    if (removedElements.length > 0) {
+      removedElements.forEach((account) => {
+        dispatch(removePermittedAccount(activeTabOrigin, account));
+      });
+    }
+
+    const newElements = selectedAccounts.filter(
+      (account) => !connectedAccountsAddresses.includes(account),
+    );
+
+    if (newElements.length > 0) {
+      dispatch(addMorePermittedAccounts(activeTabOrigin, newElements));
+    }
+  };
 
   return (
     <Modal
       isOpen
-      onClose={onClose} // Simplified inline function
+      onClose={onClose}
       data-testid="edit-accounts-modal"
       className="edit-accounts-modal"
     >
       <ModalOverlay />
       <ModalContent>
         <ModalHeader onClose={onClose}>{t('editAccounts')}</ModalHeader>
         <Box padding={4}>
```

Comment:
Hey @NidhiKJha, wrapping the body components in `ModalBody` should fix the issues here

```suggestion
        <ModalBody paddingLeft={0} paddingRight={0}>
          <Box padding={4}>
```